### PR TITLE
Proxy fee estimation feature, verify plugin core dependency change

### PR DESCRIPTION
**Pull request by FilipTxFusion**
- Add proxy fee estimation feature for the upgradable plugin
- Change verify plugin core dependency to @nomicfoundation/hardhat-verify

### CODE REVIEW

File: docs/tools/hardhat/hardhat-zksync-upgradable.md [hunk 1]
```diff
@@ -646,4 +646,65 @@ This command will verify the implementation related to the proxy, the proxy cont
...
+
+If you want to get the estimation for the beacon proxy contract, please use the `estimateGasBeaconProxy` method:
+
+``` typescript
+const totalGasEstimation = await hre.zkUpgrades.estimation.estimateGasBeacon(deployer, contract, []);
+```
+
+
+All of those methods calculate the fee for every contract in their corresponding pipeline, print that cost on the console and return the total sum.
+If you don't want to see all the separate estimations, you can use the parameter `quiet` as the last parameter in every method and receive just the returned sum.
```
Comment (dutterbutter) [inline docs/tools/hardhat/hardhat-zksync-upgradable.md:706]:
```suggestion
Each of these methods totals the fee for every contract in their respective pipeline, displays the cost on the console, and returns the cumulative sum. If you prefer not to see the individual estimations, introduce the parameter `quiet` as the final parameter in any method to receive only the returned sum.
```

**Review by dutterbutter — CHANGES_REQUESTED** [review]
Few small changes

**Comment by FilipTxFusion** [comment]
> Few small changes

Agree with all of them. Committed the changes.

**Comment by dutterbutter** [comment]
@FilipTxFusion please resolve conflicts and we should be good! Thanks 🙏

**Review by uF4No — APPROVED** [review]

**Review by dutterbutter — APPROVED** [review]

**Pull request merged**